components/TransactionPaginated.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="container">
+    <button v-if="selectedPage > 1" class="btn-move" @click="moveToBack"> < </button>
+    <ul class="paginated">
+      <li v-for="index in totalPages" :key="index" class="paginated-item">
+        <div  v-if="(index < 4 || (totalPages - index) < 3) || Math.abs(index - selectedPage) < 2">
+        <span v-if="index === (totalPages - 2) && selectedPage < (totalPages - 4)">
+          ...
+        </span>
+        <button
+            @click="changePage(index)"
+            class="paginated-button"
+            :class="{ active: selectedPage === index }"
+        >
+          {{ index }}
+        </button>
+        <span v-if="(index === 3 && selectedPage > 5)">
+                ...
+        </span>
+        </div>
+      </li>
+    </ul>
+    <button v-if="selectedPage < totalPages" class="btn-move" @click="moveToNext"> > </button>
+  </div>
+</template>
+<script>
+export default {
+  name: "transactionPaginated",
+  props: {
+    total: {
+      type: Number,
+      default: 0,
+    },
+    item: {
+      type: Number,
+      default: 25,
+    },
+    currentPage: {
+      type: Number,
+      default: 1
+    }
+  },
+data(){
+    return {
+      selectedPage: this.currentPage
+    }
+},
+  computed: {
+    totalPages() {
+      return Math.ceil(this.total / this.item);
+    },
+  },
+
+  methods: {
+    changePage(page) {
+      this.selectedPage = page;
+      this.$emit("changePage", this.selectedPage);
+    },
+    moveToBack(){
```

Comment:
![image](https://user-images.githubusercontent.com/72648845/186086836-4b053016-452b-4917-9ed7-91fd22ed1273.png)
use similar like that